apps/examples/fs_helper/fs_rw_test.c

```diff
@@ -0,0 +1,268 @@
+
+static void rw_testlog(const char *format, ...)
+{
+	va_list ap;
+
+	if (g_printopt == 'y' || g_printopt == 'Y') {
+		va_start(ap, format);
+		vfprintf(stdout, format, ap);
+		va_end(ap);
+	}
+}
+
+static int _rwt_open(char mode)
+{
+	int oflag = 0;
+
+	if (mode == 'r') {
+		oflag = O_RDONLY;
+	} else if (mode == 'w') {
+		oflag = O_WRONLY | O_CREAT | O_TRUNC;
+	} else {
+		printf("%s", USAGE);
+		exit(EXIT_FAILURE);
+	}
+
+	long t = get_time();
+	int fd = open(g_filepath, oflag);
+	if (fd == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	long dt = get_time() - t;
+	g_total_time += dt;
+	rw_testlog("open(%c):%dms, ", mode, dt);
+
+	return fd;
+}
+
+static void _rwt_close(int fd)
+{
+	long t = get_time();
+	int ret = close(fd);
+	if (ret == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	long dt = get_time() - t;
+	g_total_time += dt;
+	rw_testlog("close:%dms\n", dt);
+}
+
+static void _rwt_write(int fd, char *buf, size_t nbytes)
+{
+	int ret = 0;
+	int total_write = 0;
+
+	long t = get_time();
+	while (total_write < nbytes) {
+		ret = write(fd, buf + total_write, nbytes - total_write);
+		if (ret == 0 || ret == -1) {
+			close(fd);
+			exit(EXIT_FAILURE);
+		}
+		total_write += ret;
+	}
+	long dt = get_time() - t;
+
+	g_total_time += dt;
+	rw_testlog("write:%dms, ", dt);
+}
+
+static void _rwt_read(int fd, char *buf, size_t nbytes)
+{
+	int ret = 0;
+	int total_read = 0;
+
+	long t = get_time();
+	while (total_read < nbytes) {
+		ret = read(fd, buf + total_read, nbytes - total_read);
+		if (ret == 0 || ret == -1) {
+			close(fd);
+			exit(EXIT_FAILURE);
+		}
+		total_read += ret;
+	}
+	long dt = get_time() - t;
+
+	g_total_time += dt;
+	rw_testlog("read:%dms, ", dt);
+}
+
+static void wr_test(void *buf, size_t nbytes)
+{
+	int i = 0;
+	int fd = -1;
+	for (; i < g_iteration; i++) {
+		fd = _rwt_open('w');
+		_rwt_write(fd, buf, nbytes);
+		_rwt_close(fd);
+
+		fd = _rwt_open('r');
+		_rwt_read(fd, buf, nbytes);
+		_rwt_close(fd);
+	}
+}
+
+static void w_test(void *buf, size_t nbytes)
+{
+	int i = 0;
+	int fd = -1;
+	for (; i < g_iteration; i++) {
+		fd = _rwt_open('w');
+		_rwt_write(fd, buf, nbytes);
+		_rwt_close(fd);
+	}
+}
+
+static void r_test(char *buf, size_t nbytes)
+{
+	int fd = -1;
+	int total_write = 0;
+	int ret = 0;
+	fd = open(g_filepath, O_WRONLY | O_CREAT | O_TRUNC);
+	if (fd == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	while (total_write < nbytes) {
+		ret = write(fd, buf + total_write, nbytes - total_write);
+		if (ret == 0 || ret == -1) {
+			close(fd);
+			printf("Write Fail In %s(%d)\n", __FUNCTION__, errno);
+			exit(EXIT_FAILURE);
+		}
+		total_write += ret;
+	}
+	close(fd);
+
+	int i = 0;
+	for (; i < g_iteration; i++) {
+		fd = _rwt_open('r');
+		_rwt_read(fd, buf, nbytes);
+		_rwt_close(fd);
+	}
+}
+
+void f_rw_performance(int size, int operation)
+{
+	char buf[size];
+
+	switch (operation) {
+	case 1:
```

Comment:
How about using definitions of `TEST_OPT_READ`, `TEST_OPT_WRITE` and `TEST_OPT_WRITEREAD` instead of 1, 2 and 3 for readabiltiy?